pkg/smokescreen/metrics/metrics.go

```diff
@@ -43,143 +39,18 @@ var metrics = []string{
 	"resolver.errors_total",
 }
 
-// MetricsClient is a thin wrapper around statsd.ClientInterface. It is used to allow
-// adding arbitrary tags to Smokescreen metrics.
-//
-// MetricsClient is not thread safe and should not be used concurrently.
-type MetricsClient struct {
-	metricsTags  map[string][]string
-	statsdClient statsd.ClientInterface
-	started      atomic.Value
-}
-
 type MetricsClientInterface interface {
-	AddMetricTags(string, []string) error
+	AddMetricTags(string, map[string]string) error
 	Incr(string, float64) error
-	IncrWithTags(string, []string, float64) error
+	IncrWithTags(string, map[string]string, float64) error
 	Gauge(string, float64, float64) error
 	Histogram(string, float64, float64) error
-	HistogramWithTags(string, float64, []string, float64) error
+	HistogramWithTags(string, float64, map[string]string, float64) error
 	Timing(string, time.Duration, float64) error
-	TimingWithTags(string, time.Duration, float64, []string) error
-	StatsdClient() statsd.ClientInterface
```

Comment:
We wrap smokescreen internal with some of our other observability tooling, and we need to retain access to the underlying statsd client to pass to other internal libraries.

What would you think about keeping this in the interface (in a perfect world it wouldn't be there, but this has been there for a while) and returning nil from the prometheus client?